JWPlayerPlugin/JWPlayerViewController.m

```diff
@@ -29,6 +28,18 @@ @implementation JWPlayerViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(pause)
+                                                 name:UIApplicationDidEnterBackgroundNotification
+                                               object:nil];     // Fix for JP-5 task
+    
+}
+
+- (void)viewDidDisappear:(BOOL)animated {
+    [[NSNotificationCenter defaultCenter] removeObserver:self]; // Fix for JP-5 task
```

Comment:
1. super should be called ahead of functionality
2. In general better remove specific observer - not all.
3. View Did Disappear might be called multiple times - so please add and remove observers in a symmetric way (viewWillAppear)

Reply:
Fixed.

Reply:
Almost.
you didn't address the following the following:
`View Did Disappear might be called multiple times - so please add and remove observers in a symmetric way (viewWillAppear)`

Now you have 2 options:
1. Add the notification in a symmetric way to the removal (as I mentioned)
2. read the following: https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver and remove that line entirely and not change the place you add the notifications since we target iOS 10 and up.

Reply:
first part speaks about when you don't need to call remove observer (if you add it once and it should go away when deallocated - in this case - didLoad will happen only once)
Second part talks why you made the first change to remove a specific observer.

It's your choice to decide what's right.
I think in this case - on didLoad and remove this line is reasonable.

Reply:
I removed addObserver to viewWillAppear.